hadoop-hdds/client/src/main/java/org/apache/hadoop/ozone/client/io/ECBlockReconstructedStripeInputStream.java

```diff
@@ -578,8 +578,16 @@ protected void loadDataBuffersFromStream()
         // an IOException.
         pair.getValue().get();
       } catch (ExecutionException ee) {
-        LOG.warn("Failed to read from block {} EC index {}. Excluding the " +
-            "block", getBlockID(), index + 1, ee.getCause());
+        String message = "Failed to read from block {} EC index {}. Excluding" +
+                " the block";
+        if (LOG.isDebugEnabled() || ee.getCause() == null) {
+          LOG.debug(message, getBlockID(), index + 1, ee.getCause());
+        } else {
+          LOG.warn(message + " Exception: {} Exception Message: {}",
```

Comment:
We discussed this. if ee.getCause returns null, then we end up with NPE. Not when we may get getcause null though. Please safe guard it.

Reply:
Why do we need if else? The following should work in same log?
ee.getCause() != null ?
                ee.getCause().getClass().getName() :
                ee.getClass().getName()

Reply:
Done